app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Services\ProductService;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\Request;
+
+class ProductController extends Controller
+{
+    /**
+     * @var ProductService
+     */
+    private ProductService $productService;
+
+    /**
+     * @param ProductService $productService
+     */
+    public function __construct(ProductService $productService)
+    {
+        $this->productService = $productService;
+    }
```

Comment:
https://stitcher.io/blog/constructor-promotion-in-php-8
```suggestion
    public function __construct(private ProductService $productService)
    {
    }
```